tests/std/tests/P0896R4_ranges_alg_transform_binary/test.cpp

```diff
@@ -0,0 +1,127 @@
+
+struct binary_transform_test {
+    static constexpr std::array<P, 3> input         = {{{1, 99}, {4, 98}, {5, 97}}};
+    static constexpr std::array<double, 3> expected = {{6.4, 7.9, 8.2}};
+    static constexpr std::array<int, 3> expected    = {{7, 11, 13}};
+
+    template <class In1, class In2, class Write>
+    static constexpr void call() {
+        { // Validate range overload
+            std::array<int, 3> output{};
+            In1 wrapped_in1{input1};
+            In2 wrapped_in2{input2};
+            Write wrapped_out{output};
+
+            auto result = transform(wrapped_in1, wrapped_in2, wrapped_out, std::plus<int>{}, get_first, cast);
+            STATIC_ASSERT(same_as<decltype(result), binary_transform_result<In1, In2, Write>>);
+            assert(result.in1 == wrapped_in1.end());
+            assert(result.in2 == wrapped_in2.end());
+            assert(result.out == wrapped_out.end());
+            assert(ranges::equal(output, expected));
+        }
+        { // Validate iterator + sentinel overload
+            std::array<int, 3> output{};
+            In1 wrapped_in1{input1};
+            In2 wrapped_in2{input2};
+            Write wrapped_out{output};
+
+            auto result = transform(wrapped_in1.begin(), wrapped_in1.end(), wrapped_in2.begin(), wrapped_in2.end(),
+                wrapped_out, std::plus<int>{}, get_first, cast);
+            STATIC_ASSERT(same_as<decltype(result), binary_transform_result<In1, In2, Write>>);
+            assert(result.in1 == wrapped_in1.end());
+            assert(result.in2 == wrapped_in2.end());
+            assert(result.out == wrapped_out.end());
+            assert(ranges::equal(output, expected));
+        }
+    }
+};
+
+int main() {
+    test_in_in_write<binary_transform_test, const P, const double, int>();
+}
+
+using Elem1 = const P;
+using Elem2 = const double;
+using Elem3 = int;
+
+#ifdef TEST_EVERYTHING
+int main() {
+    // No constexpr test here; the test_in_in_write call exceeds the maximum number of steps in a constexpr computation.
+    test_in_in_write<binary_transform_test, const P, const double, int>();
+}
+#else // ^^^ test all range combinations // test only interesting range combos vvv
+template <class Elem, test::Sized IsSized>
+using fwd_test_range = test::range<forward_iterator_tag, Elem, IsSized, test::CanDifference::no, test::Common::no,
+    test::CanCompare::yes, test::ProxyRef::yes>;
+template <class Elem, test::Sized IsSized, test::Common IsCommon>
+using random_test_range = test::range<random_access_iterator_tag, Elem, IsSized, test::CanDifference::no, IsCommon,
+    test::CanCompare::yes, test::ProxyRef::no>;
+
+constexpr bool run_tests() {
+    // All (except contiguous) proxy reference types, since the algorithm doesn't really care. Cases with only 1 range
+    // sized are not interesting; common is interesting only in that it's necessary to trigger memcmp optimization.

```

Comment:
I believe that this comment was copy-pasted; `transform` doesn't have a `memcmp` optimization.

Reply:
removed